deployment/src/test/java/com/github/mcollovati/quarkus/hilla/deployment/crud/spring/ListRepositoryServiceTest.java

```diff
@@ -0,0 +1,333 @@
+
+import jakarta.inject.Inject;
+import jakarta.persistence.EntityManager;
+import java.util.List;
+
+import dev.hilla.crud.filter.AndFilter;
+import dev.hilla.crud.filter.OrFilter;
+import dev.hilla.crud.filter.PropertyStringFilter;
+import io.quarkus.builder.Version;
+import io.quarkus.maven.dependency.Dependency;
+import io.quarkus.test.QuarkusUnitTest;
+import io.quarkus.test.TestTransaction;
+import org.assertj.core.api.Assertions;
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.spec.JavaArchive;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.data.domain.Pageable;
+import org.springframework.data.domain.Sort;
+
+import static org.assertj.core.groups.Tuple.tuple;
+
+@TestTransaction
+class ListRepositoryServiceTest {
+
+    @RegisterExtension
+    static final QuarkusUnitTest config = new QuarkusUnitTest()
+            .setForcedDependencies(List.of(
+                    Dependency.of("io.quarkus", "quarkus-spring-data-jpa", Version.getVersion()),
+                    Dependency.of("io.quarkus", "quarkus-jdbc-h2", Version.getVersion())))
+            .withConfigurationResource(testResource("application.properties"))
+            .setArchiveProducer(() -> ShrinkWrap.create(JavaArchive.class)
+                    .addAsResource(testResource("import.sql"), "import.sql")
+                    .addClasses(TestEntity.class, TestRepository.class, TestListRepositoryService.class));
+
+    @Inject
+    TestListRepositoryService service;
+
+    @Inject
+    EntityManager entityManager;
+
+    @Test
+    void listService_count_nullFilter() {
+        Assertions.assertThat(service.count(null)).isEqualTo(5);
+    }
+
+    @Test
+    void listService_count_equalsFilter() {
+        PropertyStringFilter filter = new PropertyStringFilter();
+        filter.setPropertyId("text");
+        filter.setFilterValue("Two");
+        filter.setMatcher(PropertyStringFilter.Matcher.EQUALS);
+        Assertions.assertThat(service.count(filter)).isEqualTo(1);
+    }
+
+    @Test
+    void listService_count_containsFilter() {
+        PropertyStringFilter filter = new PropertyStringFilter();
+        filter.setPropertyId("text");
+        filter.setFilterValue("o");
+        filter.setMatcher(PropertyStringFilter.Matcher.CONTAINS);
+        Assertions.assertThat(service.count(filter)).isEqualTo(3);
+    }
+
+    @Test
+    void listService_count_caseInsensitiveEqualsFilter() {
+        PropertyStringFilter filter = new PropertyStringFilter();
+        filter.setPropertyId("text");
+        filter.setFilterValue("TWO");
+        filter.setMatcher(PropertyStringFilter.Matcher.EQUALS);
+        Assertions.assertThat(service.count(filter)).isEqualTo(1);
+
+        filter.setFilterValue("two");
+        Assertions.assertThat(service.count(filter)).isEqualTo(1);
+    }
```

Comment:
Whats the difference to equal filter?

Reply:
It is to test that the filter is case insensitive

Reply:
But how should it looks like if case sensitivity is required?

Reply:
Actually, it is the same behavior as in Hilla